app/commands/exercise/approach/link_matching_submissions.rb

```diff
@@ -0,0 +1,35 @@
+class Exercise::Approach::LinkMatchingSubmissions
+  include Mandate
+
+  initialize_with :approach
+
+  delegate :exercise, to: :approach
+
+  def call
+    return relink_submissions!(linked_submissions) if approach.tags.blank?
+
+    update_submissions!(linked_submissions)
+    update_submissions!(unlinked_submissions.tagged)
+  end
+
+  private
+  def linked_submissions = Submission.where(exercise:, approach:)
+  def unlinked_submissions = Submission.where(exercise:, approach: nil)
```

Comment:
These need to just be submissions that have iterations attached to them (else we'll end up doing 20x the amount of work needed). I'd probably do that via an inner join to only get records where there's an iteration with this submission_id.

Reply:
It's probably worth adding a scope on `submissions` for (`has_iteration`) or something?